utils/crypto/bls/secret.go

```diff
@@ -25,49 +25,64 @@ var (
 
 type SecretKey = blst.SecretKey
 
+type Signer interface {
+	PublicKey() *PublicKey
+	Sign(msg []byte) *Signature
+	SignProofOfPossession(msg []byte) *Signature
+	// TODO: delete me
+	ToBytes() []byte
+}
+
+var _ Signer = (*signer)(nil)
+
+type signer struct {
+	sk *SecretKey
+}
+
 // NewSecretKey generates a new secret key from the local source of
 // cryptographically secure randomness.
-func NewSecretKey() (*SecretKey, error) {
+func NewSigner() (Signer, error) {
```

Comment:
nit: export `signer` and return it as a concrete type here so that the calling code can decide whether to use it as an interface/as a concrete type. We'll have to figure out a fix/good naming scheme since it'll collide w/ the interface name as it currently is though. We could also as an alternative argue that the interface is premature and just use the struct.

Reply:
🤔  I don't think the interface is premature though, I don't plan on adding any functionality for the other implementations. What I'm trying to get is encapsulation so that you can't access the secret-key outside of this package. I think want to make a `LocalSigner`, then some "cloud-based" signers (there are a couple of options here that will each have their own concrete types).   

Having it generic from day 1 minimizes the impact of converting what's here to a `LocalSigner` (which would also be used in all tests), and other signer types would be created based on config values on application start. 

Thoughts?